packages/node/src/configure/SubqueryProject.ts

```diff
@@ -152,4 +165,58 @@ async function loadProjectTemplates(
       name: projectManifest.templates[index].name,
     }));
   }
+
+  return [];
+}
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export async function generateTimestampReferenceForBlockFilters(
+  dataSources: SubqlProjectDs[],
+  api: ApiService,
+): Promise<SubqlProjectDs[]> {
+  const cron = new Cron();
+
+  dataSources = await Promise.all(
+    dataSources.map(async (ds) => {
+      if (isRuntimeDs(ds)) {
+        const startBlock = ds.startBlock ?? 1;
+        let block: AlgorandBlock;
+        let timestampReference: Date;
+        const safeApi = api.getSafeApi(startBlock);
```

Comment:
@stwiname please review this, not sure we should use safeApi here

Reply:
We can just use the normal api here.